narayana/tools/src/main/java/io/narayana/perf/jmh/Benchmark.java

```diff
@@ -71,15 +71,40 @@ public void setPrevious(Benchmark previous) {
     @Override
     public String toString() {
         if (previous != null) {
-            double pc = ( score / previous.score - 1) * 100.0;
-            String res = regression == -1 ? "regression" : regression == 0 ? "no change" : "improvement";
-            return String.format("%s: %f vrs %f (%f%%: %s)%n",
-                    benchmark, score, previous.score, pc, res);
+            double percentChange = 0.0;
+            boolean isIntersecting = scoresIntersecting();
+            if (isIntersecting) {
+                regression = 0;
+            } else {
+                percentChange = (score / previous.score - 1) * 100.0;
+            }
+            String changeType = regression == -1 ? "regression" : regression == 0 ? "no change" : "improvement";
+            if (isIntersecting) {
+                return String.format(
+                        "%s: (score:%f error:%f) vrs (score:%f error:%f) (%f%%: %s due to score error intersection)%n",
+                        benchmark, score, scoreError, previous.score, previous.scoreError, percentChange, changeType);
+            }
+            return String.format("%s: %f vrs %f (%f%%: %s)%n", benchmark, score, previous.score, percentChange,
+                    changeType);
         } else {
             return String.format("%s: %f%n", benchmark, score);
         }
     }
 
+    /**
+     * Checks if there is any intersection between the scores of current and
+     * previous version when scoreError is considered.
+     *
+     * @return boolean
+     */
+    private boolean scoresIntersecting() {
+        if ((score - scoreError > previous.score + previous.scoreError)
+                || (score + scoreError < previous.score - previous.scoreError)) {
+            return false;
+        }
+        return true;
+    }
```

Comment:
I am not convinced that this test detects if the ranges overlap. Suppose the ranges are [x1, x2] and [y1, y2] then they will overlap if both the (start of the first interval (x1) is less than or equal to the end of the second interval (y2)) AND (start of the second interval (y1) is less than or equal to the end of the first interval (x2)). This would mean that the test should be either

        return ((score - scoreError <= previous.score + previous.scoreError)
                && (previous.score - previous.scoreError <= score + scoreError));

or equivalently:

        return ((score - scoreError > previous.score + previous.scoreError)
                !! (score - scoreError < previous.score - previous.scoreError));

Reply:
In your algorithm you are suggesting to check the overlapping condition.

But in my algorithm, I am trying the other way around (how it doesn't overlap). So in the code, if you see, if condition satisfies, it means there is no overlap and returns **false**.

> if ((score - scoreError > previous.score + previous.scoreError)
>                 || (score + scoreError < previous.score - previous.scoreError)) {
>             return false;
> }

Reply:
+1 That is an excellent point. Apologies for not spotting that.